eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -0,0 +1,254 @@
+    {
+        parent::initialize($input, $output);
+        $this->logger = $this->getContainer()->get('logger');
+        $this->searchHandler = $this->getContainer()->get('ezpublish.spi.search');
+        $this->persistenceHandler = $this->getContainer()->get('ezpublish.api.persistence_handler');
+        $this->databaseHandler = $this->getContainer()->get('ezpublish.connection');
+        if (!$this->searchHandler instanceof Indexing) {
+            throw new RuntimeException(
+                'Expected to find Search Engine Handler implementing Indexing but found something else.'
+            );
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('ezplatform:reindex')
+            ->setDescription('Recreate search engine index')
+            ->addArgument('bulk_count', InputArgument::OPTIONAL, 'Number of objects to be indexed at once', 5)
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> indexes current configured database in configured search engine index.
+EOT
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->output = $output;
+        $bulkCount = $input->getArgument('bulk_count');
+
+        $output->writeln('Creating search index for the engine: ' . get_parent_class($this->searchHandler));
+
+        $this->searchHandler->purgeIndex();
+
+        if ($this->searchHandler instanceof ContentIndexing) {
+            $this->createContentIndex($bulkCount);
+        }
+
+        if ($this->searchHandler instanceof LocationIndexing) {
+            $this->createLocationsIndex($bulkCount);
+        }
+
+        $output->writeln(PHP_EOL . 'Finished creating search index for the engine: ' . get_parent_class($this->searchHandler));
+    }
+
+    /**
+     * Wrapper for indexing Content.
+     *
+     * @param int $bulkCount a number of object rows to fetch in single batch
+     */
+    private function createContentIndex($bulkCount)
+    {
+        $query = $this->databaseHandler->createSelectQuery();
+        $query->select('count(id)')
+            ->from('ezcontentobject')
+            ->where($query->expr->eq('status', ContentInfo::STATUS_PUBLISHED));
+        $stmt = $query->prepare();
+        $stmt->execute();
+        $totalCount = $stmt->fetchColumn();
+
+        $query = $this->databaseHandler->createSelectQuery();
+        $query->select('id', 'current_version')
+            ->from('ezcontentobject')
+            ->where($query->expr->eq('status', ContentInfo::STATUS_PUBLISHED));
+
+        $stmt = $query->prepare();
+        $stmt->execute();
+
+        $this->searchHandler->purgeIndex();
+
+        $this->output->writeln('Indexing Content...');
+
+        /** @var \Symfony\Component\Console\Helper\ProgressBar $progress */
+        $progress = new ProgressBar($this->output);
```

Comment:
Here I decided to use `ProgressBar` instead of [deprecated `ProgressHelper`](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Console/Helper/ProgressHelper.php#L25). In the other commands it is retrieved using `$this->getHelperSet()->get('progress')`. Right now I don't know how to use it for `ProgressBar`.

Using `ProgressBar` also allows us to get rid of [such ugly workarounds when logging other messages](https://github.com/alongosz/ezpublish-kernel/blob/40f005e1f3e724f0c3af62ae4e412a4e033f5691/eZ/Bundle/EzPublishLegacySearchEngineBundle/Command/CreateIndexCommand.php#L176-L177)